baybe/surrogates/utils.py

```diff
@@ -57,156 +55,140 @@ def _prepare_targets(y: Tensor) -> Tensor:
     return y.to(DTypeFloatTorch)
 
 
-def catch_constant_targets(model_cls: type[Surrogate]):
-    """Wrap a ``Surrogate`` class that cannot handle constant training target values.
+def catch_constant_targets(cls: type[Surrogate], std_threshold: float = 1e-6):
+    """Make a ``Surrogate`` class robustly handle constant training targets.
+
+    More specifically, "constant training targets" can mean either of:
+        * The standard deviation of the training targets is below the given threshold.
+        * There is only one target and the standard deviation cannot even be computed.
 
-    In the wrapped class, these cases are handled by a separate model type.
+    The modified class handles the above cases separately from "regular operation"
+    by resorting to a :class:`baybe.surrogates.naive.MeanPredictionSurrogate`,
+    which is stored as an additional temporary attribute in its objects.
 
     Args:
-        model_cls: A ``Surrogate`` class that should be wrapped.
+        cls: The :class:`baybe.surrogates.base.Surrogate` to be augmented.
+        std_threshold: The standard deviation threshold below which operation is
+            switched to the alternative model.
+
+    Raises:
+        ValueError: If the class already contains an attribute with the same name
+            as the temporary attribute to be added.
 
     Returns:
-        A wrapped version of the class.
+        The modified class.
     """
+    # Name of the attribute added to store the alternative model
+    attr_name = "_constant_target_model"
```

Comment:
a slightly more descriptive variable name would be nice, this sounds like a loop variable

Reply:
you mean the left or the right side of the equation? :D
Pls provide suggestion so that I know what you mean 🙃

Reply:
changed, also in the other decorator